docs/source/en/core/unittest.md

```diff
@@ -0,0 +1,719 @@
+  },
+};
+```
+
+A corresponding test: 
+
+```js
+describe('isXHR()', () => {
+  it('should true', () => {
+    const ctx = app.mockContext({
+      headers: {
+        'X-Requested-With': 'XMLHttpRequest',
+      },
+    });
+    assert(ctx.isXHR === true);
+  });
+
+  it('should false', () => {
+    const ctx = app.mockContext({
+      headers: {
+        'X-Requested-With': 'SuperAgent',
+      },
+    });
+    assert(ctx.isXHR === false);
+  });
+});
+```
+
+### Request
+
+Extended properties and function are available on `ctx.request`, so they can be tested directly.
+
+For example, provide a `isChrome` property to `app/extend/request.js` to varify requests whether they are from Chrome or not.
+
+```js
+const IS_CHROME = Symbol('Request#isChrome');
+module.exports = {
+  get isChrome() {
+    if (!this[IS_CHROME]) {
+      const ua = this.get('User-Agent').toLowerCase();
+      this[IS_CHROME] = ua.includes('chrome/');
+    }
+    return this[IS_CHROME];
+  },
+};
+```
+
+A corresponding test: 
+
+```js
+describe('isChrome()', () => {
+  it('should true', () => {
+    const ctx = app.mockContext({
+      headers: {
+        'User-Agent': 'Chrome/56.0.2924.51',
+      },
+    });
+    assert(ctx.request.isChrome === true);
+  });
+
+  it('should false', () => {
+    const ctx = app.mockContext({
+      headers: {
+        'User-Agent': 'FireFox/1',
+      },
+    });
+    assert(ctx.request.isChrome === false);
+  });
+});
+```
+
+### Response
+
+Identical with Request, Response test could be based on `ctx.reponse` directly, accessing all the extended functions and properties.
+
+For example, provide an `isSuccess` property to indicate current status code equal to 200 or not.
+
+```js
+module.exports = {
+  get isSuccess() {
+    return this.status === 200;
+  },
+};
+```
+
+The corresponding test: 
+
+```js
+describe('isSuccess()', () => {
+  it('should true', () => {
+    const ctx = app.mockContext();
+    ctx.status = 200;
+    assert(ctx.response.isSuccess === true);
+  });
+
+  it('should false', () => {
+    const ctx = app.mockContext();
+    ctx.status = 404;
+    assert(ctx.response.isSuccess === false);
+  });
+});
+```
+
+### Helper
+
+Similiar to Service, Helper is avaliable on ctx, which can be tested directly.
+
+Such as `app/helper/format.js`:
+
+```js
+module.exports = {
+  money(val) {
+    const lang = this.ctx.get('Accept-Language');
+    if (lang.includes('zh-CN')) {
+      return `￥ ${val}`;
+    }
+    return `$ ${val}`;
+  },
+};
+```
+
+A corresponding test: 
+
+```js
+describe('money()', () => {
+  it('should RMB', () => {
+    const ctx = app.mockContext({
+      // mock headers of ctx
+      headers: {
+        'Accept-Language': 'zh-CN,zh;q=0.5',
+      },
+    });
+    assert(ctx.helper.money(100) === '￥ 100');
+  });
+
+  it('should US Dolar', () => {
+    const ctx = app.mockContext();
+    assert(ctx.helper.money(100) === '$ 100');
+  });
+});
+```
+
+## Mock 方法 
```

Comment:
Mock Functions

Reply:
@freebyron would you please to translate this one and PR will be merged.

Reply:
Done